core/src/com/unciv/ui/overviewscreen/WonderOverviewTable.kt

```diff
@@ -0,0 +1,196 @@
+
+        fun getImage() = if (status == WonderStatus.Unknown && !viewEntireMapForDebug) null
+            else category.getImage?.invoke(name, if (category == CivilopediaCategories.Terrain) 50f else 45f)
+
+        fun getNameColumn() = when {
+            viewEntireMapForDebug -> name
+            status == WonderStatus.Unknown -> status.label
+            else -> name
+        }
+
+        fun getStatusColumn() = when {
+            status != WonderStatus.Known -> status.label
+            civ == null -> status.label
+            else -> civ.civName
+        }
+
+        fun getLocationColumn() = when {
+            status <= WonderStatus.NotFound -> ""
+            location == null -> ""
+            location.isCityCenter() -> location.getCity()!!.name
+            location.getCity() != null -> "Near [${location.getCity()!!}]"
+            city != null -> "Somewhere around [$city]"
+            viewEntireMapForDebug -> location.position.toString()
+            else -> "Far away"
+        }
+    }
+
+    private val wonders: Array<WonderInfo>
+
+    init {
+        wonders = collectInfo()
+        createGrid()
+    }
+
+    private fun shouldBeDisplayed(wonder: Building) = when {
+        Constants.hideFromCivilopediaUnique in wonder.uniques -> false
+        Constants.hiddenWithoutReligionUnique in wonder.uniques && hideReligionItems -> false
+        else -> wonder.uniqueObjects.filter { unique ->
+                unique.placeholderText == "Hidden when [] Victory is disabled"
+            }.none { unique ->
+                !gameInfo.gameParameters.victoryTypes.contains(VictoryType.valueOf(unique.params[0]))
+            }
+    }
+
+    private fun collectInfo(): Array<WonderInfo> {
+        val collator = UncivGame.Current.settings.getCollatorFromLocale()
+        // Maps all World Wonders by their position in sort order to their name
+        val allWonderMap: Map<Int, String> =
+            gameInfo.ruleSet.buildings.values.asSequence()
+            .filter { it.isWonder }
+            .sortedWith(compareBy(collator, { it.name.tr() }))
+            .withIndex()
+            .map { it.index to it.value.name }
+            .toMap()
+        val wonderCount = allWonderMap.size
+
+        // Inverse of the above
+        val wonderIndexMap: Map<String, Int> = allWonderMap.map { it.value to it.key }.toMap()
+
+        // Maps all Natural Wonders on the map by name to their tile
+        val allNaturalsMap: Map<String, TileInfo> =
+            gameInfo.tileMap.values.asSequence()
+            .filter { it.isNaturalWonder() }
+            .map { it.naturalWonder!! to it }
+            .toMap()
+        val naturalsCount = allNaturalsMap.size
+
+        // Natural Wonders sort order index to name
+        val naturalsIndexMap: Map<Int, String> = allNaturalsMap.keys
+            .sortedWith(compareBy(collator, { it.tr() }))
+            .withIndex()
+            .map { it.index to it.value }
```

Comment:
.toMap accepts a mapping function iirc

Reply:
But maybe pyramids on a .addCircle white background

Reply:
Ha! .toMap() is quite rare - the only other one so far is `val roadsMap` in TileSetStrings... Did I perpetrate that, too?

... nope, there's seven versions of toMap(), four with no arguments except the generic type, and three with a destination to fill. None with any lambda in sight.